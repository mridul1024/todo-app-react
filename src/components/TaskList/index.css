.task-list-container {
  display: flex;
  width: 70vw;
  margin-left: 30%;
  justify-content: flex-start;
  align-items: center;
}

.task-list-container img {
  width: 20px;
  height: 20px;
  padding-right: 10px;
  cursor: pointer;
}

.task-list-container p {
  font-size: 18px;
  margin: 5px;
}

.complete-task {
  text-decoration: line-through;
}
.incomplete-task {
  text-decoration: none;
}
