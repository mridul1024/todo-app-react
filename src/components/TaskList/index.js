import React, { Component } from "react";
import "./index.css";

/**
 * ? Task status has two values -> 1 (complete) or 0 (incomplete)
 * * Select image based on task status -> create a function to select src -> takes taskStatus as argument
 * * Strike out tasks with completed status -> create complete-task and incomplete-task css class
 *
 * ! props -> taskStatus , task
 */

export default class TaskList extends Component {
  selectImage = (statusArgument) => {
    return statusArgument === "0"
      ? "../task_incomplete_checkbox.png"
      : "../task_complete_checkbox.png";
  };
  selectCssClass = (statusArgument) => {
    return statusArgument === "1" ? "complete-task" : "incomplete-task";
  };
  changeImage = (statusArgument) => {
    this.src = this.selectImage(statusArgument);
  };

  render() {
    return (
      <>
        <div className="task-list-container">
          <img
            onClick={this.props.changeStatus}
            src={`${this.selectImage(this.props.taskStatus)}`}
            alt="task_checkbox"
            data-testid="task-list-checkbox"
          />
          <p
            data-testid="task-list-placeholder"
            className={`${this.selectCssClass(this.props.taskStatus)}`}
          >
            {this.props.task}
          </p>
        </div>
      </>
    );
  }
}
