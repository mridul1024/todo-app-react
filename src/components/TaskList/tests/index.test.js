import React from "react";
import { screen, render, fireEvent } from "@testing-library/react";
import TaskList from "../index";

describe("<TaskList /> component tests ->", () => {
  let clickSpy;
  beforeAll(() => {
    clickSpy = jest.fn();
  });
  test("should render a TaskList component and match snapshot", () => {
    let taskListComponent;
    taskListComponent = render(
      <TaskList task="My first task" taskStatus="0" />
    );
    expect(taskListComponent.baseElement).toMatchSnapshot("baseElement");
  });

  test("should contain a incomplete-task css class in the <p> tag when status is 0", () => {
    //Arrange
    let taskPlaceholder;
    render(<TaskList task="Random task" taskStatus="0" />);
    //Act
    taskPlaceholder = screen.getByTestId("task-list-placeholder");
    //Assert
    expect(taskPlaceholder.getAttribute("class")).toBe("incomplete-task");
  });

  test("should not contain a complete-task css class in the <p> tag when status is 0", () => {
    let taskPlaceholder;
    render(<TaskList task="Random task" taskStatus="0" />);
    taskPlaceholder = screen.getByTestId("task-list-placeholder");
    expect(taskPlaceholder.getAttribute("class")).not.toBe("complete-task");
  });

  test("should contain a complete-task css class in the <p> tag when the status is 1", () => {
    let taskPlaceholder;
    render(<TaskList task="Random task" taskStatus="1"></TaskList>);
    taskPlaceholder = screen.getByTestId("task-list-placeholder");
    expect(taskPlaceholder.getAttribute("class")).toBe("complete-task");
  });

  test("should not contain a incomplete-task css class in the <p> tag when status is 1", () => {
    let taskPlaceholder;
    render(<TaskList task="Random task" taskStatus="1" />);
    taskPlaceholder = screen.getByTestId("task-list-placeholder");
    expect(taskPlaceholder.getAttribute("class")).not.toBe("incomplete-task");
  });

  test("should call prop function tickTask on click", () => {
    let imageComponent;
    render(
      <TaskList task="Random task" taskStatus="1" changeStatus={clickSpy} />
    );
    imageComponent = screen.getByTestId("task-list-checkbox");
    fireEvent.click(imageComponent);
    expect(clickSpy).toBeCalled();
  });

  test("should have an unfilled checkbox as img src when status is 0", () => {
    let img;
    render(
      <TaskList task="Random task" taskStatus="0" changeStatus={clickSpy} />
    );
    img = screen.getByTestId("task-list-checkbox");
    expect(img.getAttribute("src")).toBe("../task_incomplete_checkbox.png");
  });

  test("should have a filled checkbox as img src when status is 1", () => {
    let img;
    render(
      <TaskList task="Random task" taskStatus="1" changeStatus={clickSpy} />
    );
    img = screen.getByTestId("task-list-checkbox");
    expect(img.getAttribute("src")).toBe("../task_complete_checkbox.png");
  });
});
