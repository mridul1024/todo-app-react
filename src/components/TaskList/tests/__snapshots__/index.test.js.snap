// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskList /> component tests -> should render a TaskList component and match snapshot: baseElement 1`] = `
<body>
  <div>
    <div
      class="task-list-container"
    >
      <img
        alt="task_checkbox"
        data-testid="task-list-checkbox"
        src="../task_incomplete_checkbox.png"
      />
      <p
        class="incomplete-task"
        data-testid="task-list-placeholder"
      >
        My first task
      </p>
    </div>
  </div>
</body>
`;
